libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/Zynq/uncached_memory.c

```diff
@@ -159,29 +137,29 @@ uint32_t ulSkipSize;
 
 static void vInitialiseUncachedMemory( )
 {
-extern u8 _end;
-
 	/* At the end of program's space... */
-	pucStartOfMemory = (uint8_t *) &_end;
+	pucStartOfMemory = ( uint8_t * ) &( _end );
 
 	/* Align the start address to 1 MB boundary. */
-	pucStartOfMemory = (uint8_t *)( ( ( uint32_t )pucStartOfMemory + UNCACHED_MEMORY_SIZE ) & ( ~( UNCACHED_MEMORY_SIZE - 1 ) ) );
+	pucStartOfMemory = ( uint8_t * )( ( ( uint32_t )pucStartOfMemory + uncMEMORY_SIZE ) & ( ~( uncMEMORY_SIZE - 1 ) ) );
 
-	if( ( ( u32 )pucStartOfMemory ) + UNCACHED_MEMORY_SIZE > DDR_MEMORY_END )
+	if( ( ( u32 )pucStartOfMemory ) + uncMEMORY_SIZE > DDR_MEMORY_END )
 	{
-		vLoggingPrintf( "vInitialiseUncachedMemory: Can not allocate uncached memory\n" );
+		FreeRTOS_printf( ( "vInitialiseUncachedMemory: Can not allocate uncached memory\n" ) );
 	}
 	else
 	{
 		/* Some objects want to be stored in uncached memory. Hence the 1 MB
 		address range that starts after "_end" is made uncached by setting
 		appropriate attributes in the translation table. */
-		Xil_SetTlbAttributes( ( uint32_t )pucStartOfMemory, 0xc02 ); /* addr, attr */
+		/* FIXME claudio rossi. Modified to prevent data abort exception (misaligned access)
```

Comment:
Is this `FIXME` still accurate?  If so, can we eliminate it in this revision?

Reply:
> Is this FIXME still accurate?

In fact, nothing has to be fixed. A developer Claudio Rossi proposed to change the memory attribute from `0xC02` to `0x1C02`:

`#define uncMEMORY_ATTRIBUTE       0x1C02`

I do not know what is the meaning of the extra bit `0x1000`, but it works equally well.

I don't mind removing the text.

Reply:
> I do not know what is the meaning of the extra bit `0x1000`

Looking at the earlier comment: it allows unaligned access in the specified memory range.

> @gkwicker requested changes 

Are those changes now satisfactory?

Anyone else wants to review this PR? 
@yuhui-zheng maybe?

Note, this PR received the wrong title. It should have bee something like:
**FreeRTOS+TCP : Revision of Xilinx Network Interface Version 2**